web/packages/teleterm/src/sharedProcess/ptyHost/ptyProcess.ts

```diff
@@ -76,8 +82,7 @@ export class PtyProcess extends EventEmitter implements IPtyProcess {
         // https://unix.stackexchange.com/questions/123858
         cwd: this.options.cwd || getDefaultCwd(this.options.env),
         env: this.options.env,
-        // Turn off ConPTY due to an uncaught exception being thrown when a PTY is closed.
-        useConpty: false,
+        useConpty: this.options.useConpty,
```

Comment:
Hmm, every time I open a new shell, there's a new `conhost.exe` under Task Manager -> Details. If I explicitly close a tab, it goes away. But when I just close Connect, that `conhost.exe` never goes away.

Reply:
That isn't the case with winpty, where a new conhost.exe is created, but it goes away when closing Connect.

Reply:
I'm trying to reproduce it, but with no luck. 
In the recording below, `conhost.exe` process are killed both when closing tabs one by one and when closing the entire app:

https://github.com/user-attachments/assets/5eb32d9d-32f5-4002-a09e-c5cb86b17a65

I also don't see these extra arguments:
<img width="890" alt="image" src="https://github.com/user-attachments/assets/db34fd3b-9998-4252-9c12-72cd282f4b62">

Reply:
I've just tried your build. conhost.exe goes away when starting a local terminal. But when connecting through SSH to a node, it stays open after closing Connect.

I've tested this only with SSH nodes. Because of the bug with the `css` prop, I cannot access additional action menu and because the shortcut to open a new terminal tab doesn't work inside the VM, I couldn't open a local terminal. ;f

Reply:
I'd look if there are any extra node-pty options there to take care of the issue or if someone else had the same problem. It's interesting that this happens with ConPTY but not winpty.

Reply:
I didn't find any `node-pty` options that would take care of that. When it comes to similar issues, I found some (https://github.com/microsoft/node-pty/issues/333, https://github.com/Eugeny/tabby/issues/6318). 
The most useful seems this PR from vscode https://github.com/microsoft/vscode/pull/118437.  

However, it seems to me that simply killing all pty process on exit would work. I tested this by setting a 30 seconds timeout after the shared process launched; after that, I didn't see any remaining `conhost.exe` processes.
The problem is that `process.on('exit')` doesn't seem to work on Windows and our cleanup code is not invoked at all :/ I think we need to send a graceful kill message to the shared process in a different way.

Reply:
I have it 🥵
To make it work I needed two things:
1. Gracefully killing the shared process on Windows (done with `process.send`).
2. Waiting for the pty processes to exit before closing the app. Killing the pty process takes some time (from my testing, the time between calling `ptyProcess.kill` and `ptyProcess.onExit` was around 400 ms). If I'm correct, in vsocde they set timeouts to achieve waiting for processes to exit, but I thought that the better option is to simply wait for `onExit` to be called (I took an inspiration from `terminateWithTimeout`).

Tested on Windows with ConPTY enabled/disabled and on macOS. I didn't see any unterminated processes.


https://github.com/user-attachments/assets/6c7fdb61-544c-4092-bda0-7ecb198c7452